src/lib/source-handlers/gitlab/list-repos.ts

```diff
@@ -29,7 +29,10 @@ export async function fetchGitlabReposForPage(
     hasNextPage = true;
     repoData.push(
       ...projects
-        .filter((project: any) => !project.archived)
+        .filter(
+          (project: any) =>
+            !project.archived && project.shared_with_groups.length == 0,
```

Comment:
is this a safe assumption that `shared_with_groups` is always present? If not please do a safe check `project.shared_with_groups?.length == 0`

Reply:
Done